### ISSUE TITLE
waitForCompletion(value) breaks ( fails ) in 1.1.2-SNAPSHOT

### ISSUE THREAD

**Issue by amprasanna** [eclipse-paho/paho.mqtt.java#386]
Please fill out the form below before submitting, thank you!

- [ ] Bug exists Release Version 1.1.1 ( Master Branch)
- [ X] Bug exists in Snapshot Version 1.1.2-SNAPSHOT (Develop Branch)

Issue:
I'm currently making use of the 1.1.2-SNAPSHOT. While leveraging it's recent additions, i'm trying out to test/simulate network failures using the [ConnectionManipulationProxyServer.java](https://github.com/eclipse/paho.mqtt.java/blob/master/org.eclipse.paho.client.mqttv3.test/src/test/java/org/eclipse/paho/client/mqttv3/test/utilities/ConnectionManipulationProxyServer.java).
As part of the test case, when I use "waitForCompletion(value)", where value is any Integer value (ex: 1000 * 10), the execution breaks immediately when it reaches the 'waitForCompletion(value)'.

Basically, it doesn't wait nor behaves as defined, while executing the waitForCompletion(value), but breaks/squashes the execution.

Downgrading the Paho MQTT version back to 1.1.1 (Release build), the testing scenario behaves in the intended manner without any changes to the code set.
